### ISSUE TITLE
[data-table] docs outdated since style guide refactor

### ISSUE THREAD

**Issue by kerim1510** [material-components/material-components-web#6169]
## Bug report
The core styles for data-table is an undefined mixin.

### Steps to reproduce
When compiling the application with webpack.

### Actual behavior
The core styles for data-table is an undefined mixin.

### Screenshots
![image](https://user-images.githubusercontent.com/12524605/86152244-32820f80-bb00-11ea-9fad-6eab80ec5239.png)

### Your Environment:
| Software         | Version(s) |
| ---------------- | ---------- |
| MDC Web          | 7.0 |
| Browser          | Not relevant |
| Operating System | MacOS Catalina Version 10.15.5 |

**Comment by allan-chen** [comment]
Hi Kerim,

Thanks for reaching out. Are you trying to call the `core-styles` mixin yourself? There was a recent SASS refactor that may have caused this.

**Comment by kerim1510** [comment]
Hi Allan,

I had copied the code from the documentation (https://material.io/develop/web/components/data-tables)
But since version 7.0.0 this stopped working.

I don't need the core-styles mixin, so it's not a problem anymore.
I now load the styling by `@use "@material/data-table/mdc-data-table";`

**Comment by allan-chen** [comment]
I see, thanks! This appears to be a docs issue. We've recently adopted a new style guide and data table was one of the first to  follow through with it. We'll work on updating the docs to match.

**Comment by brettsheffield** [comment]
Also affects icon-buttons.  Various docs (eg. https://github.com/material-components/material-components-web/blob/master/packages/mdc-icon-button/README.md) refer to:
  `@include icon-button.core-styles;`
which no longer works.  The following works as expected:
  `@import "@material/icon-button/mdc-icon-button";`

Also, quite a lot of the docs refer to the deprecated sass `@import`, which should perhaps be updated to `@use`

**Comment by brettsheffield** [comment]
NB: This is more than just a documentation problem. It's a bug. Any apps using the old imports are broken and need updating due to this backwards-incompatible change.

**Comment by allan-chen** [comment]
Hi Brett,

The icon buttons doc should be current. The `index` Sass file should take care of forwarding the `core-styles` mixin to the icon button module. Please ensure you have the latest version (v7.0 as of today) installed. If it still fails to build, we can investigate with the error message.

I did a quick search and most of the docs have already been converted to `@use` syntax. The outliers are `layout-grid` and `typography`. We'll get those two updated.

**Comment by brettsheffield** [comment]
I have the latest version 7.0.0 of all components.

**Comment by brettsheffield** [comment]
Using `@include icon-button.core-styles;` I get the following error:
```
SassError: There is no module with the namespace "icon-button".
   ╷
26 │ @include icon-button.core-styles;
```

Works as expected with:

`@import "@material/icon-button/mdc-icon-button";`

**Comment by allan-chen** [comment]
Sounds like the module is not importing correctly. Are you importing the module using `@use @material/icon-button`?

For your reference, here's Sass in a working downstream repository that applies style icon button:
https://github.com/material-components/material-components-web-components/blob/master/packages/icon-button/src/mwc-icon-button.scss#L24

**Comment by brettsheffield** [comment]
Thanks allan-chen.  Will take a look at that.

Am importing using @import (presently).  The tutorials at https://material.io/collections/developer-tutorials/#web refer to @import rather than @use, as do the sample git repos.

I tried converting the examples to work with @use, but that caused problems with getting my color themes to work, so I went back to @import as per the developer tutorials.

I'm a new user trying out MDC Web with a view to possibly using it in a project and the documentation has been rather challenging to follow.  Or even to find.  The most useful information appears to be in the README files of the git repos, but getting there from the designer-focused https://material.io/ site was a very long and frustrating journey, and still required some hackery to get working.

**Comment by brettsheffield** [comment]
Hmm.  That repo ( https://github.com/material-components/material-components-web-components/blob/master/packages/icon-button/src/mwc-icon-button.scss#L24 ) uses 
`@include iconbutton.core-styles();`, not
`@include icon-button.core-styles;`

NB: lack of hypen in namespace and () at the end.

**Comment by allan-chen** [comment]
Don't worry about that part - it's because it aliases the module in the line `@use '@material/icon-button' as iconbutton;`.

In general, if it's `@use '@material/icon-button' as foo;` then the corresponding usage would be `foo.core0styles();`

**Comment by allan-chen** [comment]
Thank you for providing more context. With regard to `material.io` vs READMEs, the READMEs are the most current source of information for the current state of the repo, and unfortunately material.io and the codelabs can lag behind due to the release cycle schedules. We do work on periodically updating these codelabs - I have filed #6178 to track the docs issue so we can keep this focused on datatables.

Side note: sometimes the READMEs can be *too* new, especially if there have been unreleased major changes. In general the NPM package page for the release tag you're using should have the most current information, i.e. https://www.npmjs.com/package/@material/icon-button

**Comment by youbet1980** [comment]
I am still having the problem with data-table trying to follow the documentation here: https://material.io/develop/web/components/data-tables

`ERROR in ./app.scss
Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: Undefined mixin.
   ╷
52 │ @include data-table.core-styles;
   │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^`

 line is in package.json
`    "@material/data-table": "^7.0.0",
`
Showing that it is using the latest release.

My code in app.scc relating to data table. If I delete both of the last 2 lines it works but then I don't have the core styles to use.

`
@use "@material/data-table";
@include data-table.core-styles;
@include data-table.theme-baseline;
`
@allan-chen or @kerim1510 did you have any luck getting a resolution to this?

**Comment by abhiomkar** [comment]
@youbet1980 Please see [updated README](https://github.com/material-components/material-components-web/blob/master/packages/mdc-data-table/README.md) file for correct usage.

```scss
@use "@material/checkbox"; // Required only for data table with row selection.
@use "@material/icon-button"; // Required only for data table with column sorting.
@use "@material/data-table/data-table";

@include checkbox.core-styles;
@include icon-button.core-styles;
@include data-table.core-styles;
@include data-table.theme-baseline;
```

**Comment by youbet1980** [comment]
@abhiomkar Yes I Have added all these in. ( See my screenshot )
<img width="594" alt="Screen Shot 2020-08-14 at 10 28 00 am" src="https://user-images.githubusercontent.com/48540186/90199747-f6d5a780-de18-11ea-904f-1b3baedae3ee.png">

I still get the error.
<img width="487" alt="Screen Shot 2020-08-14 at 10 30 00 am" src="https://user-images.githubusercontent.com/48540186/90199844-2ab0cd00-de19-11ea-800d-42b4ead7c89c.png">

**Comment by abhiomkar** [comment]
You're missing `/data-table` in the import line.

`@use "@material/data-table";` => `@use "@material/data-table/data-table";`

**Comment by youbet1980** [comment]
@abhiomkar Now it is working. You made my day. Thank you.

**Comment by abhiomkar** [comment]
Fixed in #6317
